<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QMainWindow" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>650</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1000</width>
    <height>650</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1000</width>
    <height>650</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>SoftRenderer</string>
  </property>
  <property name="windowIcon">
   <iconset theme="QIcon::ThemeIcon::Computer"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="RenderWidget" name="renderWidget" native="true">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>800</width>
      <height>600</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>800</width>
      <height>600</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>800</width>
      <height>600</height>
     </size>
    </property>
   </widget>
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>0</y>
      <width>200</width>
      <height>150</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>200</width>
      <height>150</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>200</width>
      <height>150</height>
     </size>
    </property>
    <property name="frameShape">
     <enum>QFrame::Shape::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Shadow::Raised</enum>
    </property>
    <widget class="QLabel" name="TriangleNumberlabel">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>0</y>
       <width>131</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <bold>false</bold>
      </font>
     </property>
     <property name="text">
      <string>TriangleNumber : </string>
     </property>
    </widget>
    <widget class="QLabel" name="TriangleNumber">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>30</y>
       <width>131</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="VertexNumberlabel">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>70</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <bold>false</bold>
      </font>
     </property>
     <property name="text">
      <string>VertexNumber : </string>
     </property>
    </widget>
    <widget class="QLabel" name="VertexNumber">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>110</y>
       <width>131</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>150</y>
      <width>201</width>
      <height>451</height>
     </rect>
    </property>
    <widget class="QDial" name="Yawdial">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>370</y>
       <width>81</width>
       <height>71</height>
      </rect>
     </property>
    </widget>
    <widget class="QSlider" name="PitchSlider">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>370</y>
       <width>41</width>
       <height>71</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Orientation::Vertical</enum>
     </property>
    </widget>
    <widget class="QLabel" name="Nearlabel_3">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>320</y>
       <width>91</width>
       <height>61</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>11</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Yaw</string>
     </property>
    </widget>
    <widget class="QLabel" name="Nearlabel_2">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>320</y>
       <width>91</width>
       <height>61</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Pitch</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="MeshcheckBox">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>0</y>
       <width>131</width>
       <height>51</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>11</pointsize>
       <bold>false</bold>
      </font>
     </property>
     <property name="text">
      <string>ShaderMesh</string>
     </property>
     <property name="iconSize">
      <size>
       <width>16</width>
       <height>16</height>
      </size>
     </property>
     <property name="autoRepeatDelay">
      <number>300</number>
     </property>
     <property name="tristate">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QCheckBox" name="VertexCheckBox">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>40</y>
       <width>131</width>
       <height>51</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <bold>false</bold>
      </font>
     </property>
     <property name="text">
      <string>ShaderVertex</string>
     </property>
     <property name="shortcut">
      <string notr="true"/>
     </property>
    </widget>
    <widget class="QCheckBox" name="RasterizationCheckBox">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>80</y>
       <width>131</width>
       <height>51</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>11</pointsize>
       <bold>false</bold>
      </font>
     </property>
     <property name="text">
      <string>Rasterization</string>
     </property>
     <property name="iconSize">
      <size>
       <width>16</width>
       <height>16</height>
      </size>
     </property>
     <property name="autoRepeatDelay">
      <number>300</number>
     </property>
     <property name="tristate">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLabel" name="Nearlabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>260</y>
       <width>91</width>
       <height>61</height>
      </rect>
     </property>
     <property name="text">
      <string>Near</string>
     </property>
    </widget>
    <widget class="QSlider" name="NearSlider">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>260</y>
       <width>121</width>
       <height>61</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="Fovlabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>200</y>
       <width>91</width>
       <height>61</height>
      </rect>
     </property>
     <property name="text">
      <string>Fov</string>
     </property>
    </widget>
    <widget class="QSlider" name="FovSlider">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>200</y>
       <width>121</width>
       <height>61</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
    </widget>
    <widget class="QCheckBox" name="checkBox">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>150</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>stop</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1000</width>
     <height>27</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionopen_file"/>
    <addaction name="actionsave_image"/>
    <addaction name="actionexit"/>
   </widget>
   <widget class="QMenu" name="menuSetting">
    <property name="title">
     <string>Setting</string>
    </property>
    <widget class="QMenu" name="menuMultiThread">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="title">
      <string>ThreadMode</string>
     </property>
     <addaction name="actionMultiThread"/>
     <addaction name="actionTbbMultiThread"/>
    </widget>
    <addaction name="menuMultiThread"/>
    <addaction name="actionFaceCulling"/>
    <addaction name="actionSIMD"/>
    <addaction name="actionTexture"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuSetting"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionopen_file">
   <property name="text">
    <string>Open File</string>
   </property>
  </action>
  <action name="actionsave_image">
   <property name="text">
    <string>Save Image</string>
   </property>
  </action>
  <action name="actionexit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionFaceCulling">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>FaceCulling</string>
   </property>
  </action>
  <action name="actionSIMD">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>SIMD</string>
   </property>
  </action>
  <action name="actionTexture">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Texture</string>
   </property>
  </action>
  <action name="actionMultiThread">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>MultiThread</string>
   </property>
  </action>
  <action name="actionTbbMultiThread">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>TbbMultiThread</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>RenderWidget</class>
   <extends>QWidget</extends>
   <header>RenderWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
